src/main/java/com/salesforce/dynamodbv2/mt/mappers/MtAmazonDynamoDbStreamsByTable.java

```diff
@@ -17,9 +23,31 @@
     }
 
     @Override
-    protected Function<Record, MtRecord> getRecordMapper(StreamArn arn) {
-        String[] tenantAndTableName = mtDynamoDb.getTenantAndTableName(arn.getTableName());
-        return record -> mapRecord(tenantAndTableName[0], tenantAndTableName[1], record);
+    protected MtGetRecordsResult getMtRecords(GetRecordsRequest request, StreamArn streamArn) {
+        final String[] tenantAndTableName = mtDynamoDb.getTenantAndTableName(streamArn.getTableName());
+        return getMtRecords(request, tenantAndTableName[0], tenantAndTableName[1]);
+    }
```

Comment:
If by table is called for physical table, it derives the tenant context and table name from the physical table.